test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -1731,6 +1731,20 @@ public static boolean indexExists(String index, Client client) {
         return getIndexResponse.getIndices().length > 0;
     }
 
+    public static void awaitIndexExists(String index) {
+        awaitIndexExists(index, client());
+    }
+
+    public static void awaitIndexExists(String index, Client client) {
+        safeGet(
+            client.admin()
+                .cluster()
+                .prepareHealth(SAFE_AWAIT_TIMEOUT, index)
+                .setIndicesOptions(IndicesOptions.LENIENT_EXPAND_OPEN_CLOSED)
+                .execute()
+        );
```

Comment:
Is there a reason we're not using something like `ESIntegTestCase#awaitClusterState` here? We're not waiting for a specific health (i.e. `GREEN`) here, just for the index to exist. Couldn't (/shouldn't) we do that with a predicate?

Reply:
Really just that this is replacing pre-existing client-based calls. It's the same either way really, the health API waits with a `ClusterStateObserver`.

Reply:
One difference is that this method is also used to [check a remote cluster's indices](https://github.com/elastic/elasticsearch/pull/126501/files#diff-4df458b6c55784e819fedf7a8471e10446522e45f2c57e5dbdc52db7640c3c8eR121), i.e. it needs a client.

The other difference that I just noticed from the CI failure is that index name can be a wildcard like this [usage](https://github.com/elastic/elasticsearch/blob/4ea4a298a48521968fafd2b3f0a7174d2099cf8c/x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/MultiNodesStatsTests.java#L74). It requires expansion. Unfortunately, this means we have to use assertBusy in this case. I pushed 75c14647f00a5aad5b5acadeb53c730310cb9542

Reply:
The remote cluster in question is still running in the test JVM so no this doesn't _need_ a client, you could use `awaitClusterState` with the `ClusterService` of the elected master of the follower cluster obtained with `getFollowerCluster().getCurrentMasterNodeInstance(ClusterService.class)`.

I don't think we should support wildcards in this utility, especially if it requires an `assertBusy` like that. The only test that uses it is pretty odd (and very old). I'd rather we did something like https://github.com/elastic/elasticsearch/pull/126582 there.